plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchRequestProcessor.java

```diff
@@ -376,23 +392,12 @@ public void onFailure(Exception e) {
         }, Math.max(inputMapSize, 1));
     }
 
-    /**
-     * Validates that the query fields specified in the input and output mappings exist in the query string.
-     * @param processInputMap  the list of input mappings
-     * @param processOutputMap the list of output mappings
-     * @param queryString      the query string to be validated
-     * @return true if all query fields exist in the query string, false otherwise
-     */
-    private boolean validateQueryFieldInQueryString(
-        List<Map<String, String>> processInputMap,
-        List<Map<String, String>> processOutputMap,
-        String queryString
-    ) {
+    private boolean validateRequiredInputMappingFields(List<Map<String, String>> processInputMap, String queryString) {
```

Comment:
This function `validateRequiredInputMappingFields` has same declare and very similar implementation with redundant code as the next function  `validateRequiredOutputMappingFields`. Could we abstract another function to implement the common logic?

Reply:
we could but the logic is different for input map and output map, for example, if there is a query_template then we can skip checking output mapping field in query, it's easier to understand and faster return. If we abstract the codes, it makes it difficult to read and maintain

Reply:
Sounds good, thanks for the explanation.